lib/rubocop/cop/style/redundant_parentheses.rb

```diff
@@ -5,6 +5,13 @@ module Cop
     module Style
       # Checks for redundant parentheses.
       #
+      # Parentheses are allowed in the following cases for readability:
```

Comment:
Looking at the tests we've allowed them everywhere where there is assignment, and I don't think this always improves the readability. 

I still think that by default they should allowed only where removing them would change the AST, and that there should be an option to allow them in assignments more broadly.